src/internal/operators/take.ts

```diff
@@ -45,26 +51,41 @@ import { OperatorSubscriber } from './OperatorSubscriber';
  * the source if the source emits fewer than `count` values.
  */
 export function take<T>(count: number): MonoTypeOperatorFunction<T> {
-  return count <= 0
-    ? // If we are taking no values, that's empty.
-      () => EMPTY
-    : operate((source, subscriber) => {
-        let seen = 0;
-        source.subscribe(
-          new OperatorSubscriber(subscriber, (value) => {
-            // Increment the number of values we have seen,
-            // then check it against the allowed count to see
-            // if we are still letting values through.
-            if (++seen <= count) {
-              subscriber.next(value);
-              // If we have met or passed our allowed count,
-              // we need to complete. We have to do <= here,
-              // because re-entrant code will increment `seen` twice.
-              if (count <= seen) {
-                subscriber.complete();
-              }
-            }
-          })
-        );
-      });
+  // this condition also returns true on left-side NaN and not just if count
+  // is smaller than one. in both cases we take no value and that is the same as
+  // EMPTY
+  if (!(count >= 1)) {
+    return () => EMPTY;
+  }
+
+  // this condition also returns true on left-side Infinity. the runtime can not
+  // count beyond Number.MAX_SAFE_INTEGER so "seen" would be stuck at
+  // Number.MAX_SAFE_INTEGER + 1, which would take everything from source
+  // anyway. therefore we can just return the source and avoid the useless
+  // counting
+  if (count >= Number.MAX_SAFE_INTEGER + 1) {
+    return identity;
+  }
```

Comment:
this entire check is not needed. the behavior of `operate` below will be the same as `identity` if `count >= Number.MAX_SAFE_INTEGER + 1` (including Infinity). we probably don't expect to hit this intentionally with high numbers, but using `take(condition ? n : Infinity)` could be a valid pattern. in any way, we don't need to count seen in these cases.